Merge Request

Tamina15 opened the pull request:
Implement Upload Image To Cloudinary
Implement Validation and Update API Endpoints
Implement Authorization using Role base and Authentication using JWT
Implement Swagger

Tamina15 committed:
Update Util.java

Tamina15 committed:
Upload Image to Cloudinary (Experiment)

Tamina15 committed:
Update API endpoints and Implement Validation

Tamina15 committed:
Validation Error and Cart Error (Commit a6aecc9)

Tamina15 committed:
Change Relationship to OneToMany

Change Relationship between Product and Category from ManyToMany to OneToMany with join table

Tamina15 committed:
Add Authentication and Authorization

Add Authentication using JWT and simple Authorization using Role Base

Tamina15 committed:
Authentication and Authorization 2

Second commit

Tamina15 committed:
Change Folder

Tamina15 committed:
Implement Swagger

Remove Spring Fox Swagger
Change to OpenAPI

Tamina15 committed:
Update .gitignore

MYDINHLT1 commented on src/main/java/rookiesspring/configuration/SecurityConfiguration.java:
```diff
@@ -0,0 +1,75 @@
+                                .requestMatchers(HttpMethod.GET, "/api/v1/orders/**").hasRole("USER")
+                                .requestMatchers(HttpMethod.GET).permitAll()
+                                .requestMatchers(HttpMethod.GET, "/api/v1/users/info").hasRole("USER")
+                                .anyRequest().hasRole("USER")
```
You should define roles as constants to avoid causing mistakes

MYDINHLT1 commented on src/main/java/rookiesspring/controller/ProductController.java:
```diff
@@ -19,25 +24,31 @@
     }
 
     @GetMapping({"", "/", "/all"})
-    public ResponseEntity getAllProduct( @RequestParam(required = false, value = "name") String name,
```
Please help to update APIs I mentioned before to follow the restful API standard. Eg: /all, /full

MYDINHLT1 commented on src/main/java/rookiesspring/controller/ProductController.java:
```diff
@@ -67,53 +78,74 @@ public ResponseEntity addProduct(@RequestBody() ProductDTO product) {
 }
 
-record AddImageDTO(long product_id, ImageDTO[] images) {
+record AddImageDTO(
```
Why do you put this code here? It should be put at DTO instead of controller

MYDINHLT1 commented on src/main/java/rookiesspring/model/Product.java:
```diff
@@ -48,14 +49,18 @@ public class Product extends AuditEntity<Long> {
     @ColumnDefault(value = "'none'")
     private String rating;
```
Please help to update the data type properly

Tamina15 committed:
Split to Admin Controller and Normal User Controller

Remove POST, PUT, PATCH, DELETE methods from Normal Controller

Tamina15 committed:
Comment Out Import Bill Related File

Tamina15 committed:
Update SecurityConfiguration.java

Change to Role enum

Tamina15 committed:
Combine findAll function

+ Reformat and Cleanup

Tamina15 committed:
Combine findAll function in ProductService #2

Change @RequestParam to ProductRequestDTO class
Change rating data type
Increase return fields

Tamina15 committed:
Reformat and Cleanup

MYDINHLT1 commented on src/main/java/rookiesspring/controller/admin/CategoryAdminController.java:
```diff
@@ -0,0 +1,89 @@
+
+    @GetMapping("/products")
+    public ResponseEntity getAllCategoriesWithProducts(@RequestParam(required = false) String name) {
+        return ResponseEntity.ok(service.findAllFull(name));
```
You need to pay attention to naming functions so that they are clear and meaningful, eg: findAllFull, findByIdFull,…

MYDINHLT1 commented on src/main/java/rookiesspring/controller/admin/SupplierController.java:
```diff
@@ -0,0 +1,60 @@
+        return service.findAllByName(name);
+    }
+
+    @GetMapping("/full")
```
You should follow restful api name rules

Tamina15 committed:
Change from id to User entity

Tamina15 committed:
Implement Product Rating

Tamina15 committed:
Cache Category ID

Tamina15 committed:
Update Authentication and Authorization

Tamina15 committed:
Implement Soft Delete. Continuing

Tamina15 committed:
Implement Upload Image

Tamina15 committed:
Update products

Includes sorting and paging

Tamina15 committed:
Update JWT

Tamina15 committed:
Implement Cart

Tamina15 committed:
Seperate Rate from Product

Tamina15 committed:
Implement Update Category

+ Clean up

Tamina15 committed:
Implement Soft Delete

Seperate Image from Product Service

Tamina15 committed:
Implement Soft Delete, Block

Tamina15 committed:
Remove checkExist from Service

Tamina15 committed:
Implement updateCategories in Product

Replacement of Add and Remove Categories methods from Product

Tamina15 committed:
Remove MapStruct

Tamina15 committed:
Remove MapStruct, checkExist. + Reformat

Tamina15 committed:
Unit Test

Tamina15 committed:
Restructure File

Tamina15 committed:
Remove Unused Methods

Tamina15 committed:
Implement updateCategories (2)

Tamina15 committed:
Add h2 database for testing

Tamina15 committed:
Add JaCoCo

Tamina15 committed:
Update Testing

Tamina15 committed:
Reformat And Clean Up Code

Tamina15 committed:
Order And Cart

Tamina15 committed:
Tesing JPA Specification on Product and User

Tamina15 committed:
Merge branch 'main' into dev
